core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -197,14 +200,16 @@ public String toString() {
                 List<MetadataLogEntry> previousFiles) {
     Preconditions.checkArgument(formatVersion <= SUPPORTED_TABLE_FORMAT_VERSION,
         "Unsupported format version: v%s", formatVersion);
-    if (formatVersion > 1) {
-      Preconditions.checkArgument(uuid != null, "UUID is required in format v%s", formatVersion);
-    }
+    Preconditions.checkArgument(formatVersion == 1 || uuid != null,
+        "UUID is required in format v%s", formatVersion);
+    Preconditions.checkArgument(formatVersion > 1 || lastSequenceNumber == 0,
+        "Sequence number must be 0 in v1: %s", lastSequenceNumber);
```

Comment:
So, v1 writers will only be allowed to write sequence numbers as 0 and we will assume 0 as sequence numbers in metadata entities without an explicit value, correct?

Reply:
V1 writers should not pass any sequence number other than 0, and it won't get written to the file even though it is compatible. And reading v1 metadata should always fill in the sequence number with 0.